engine/src/gnuhash.h

```diff
@@ -2,63 +2,22 @@
 #define _GNUHASH_H_
 #include "config.h"
 #include <unordered_map>
+#include <functional>
 
 #define vsUMap     std::unordered_map
 #define vsHashComp std::hash_compare
 #define vsHash     std::hash
-class Unit;
 
 namespace std
 {
-template < >
-class hash< void* >
```

Comment:
Based on some warnings I'm seeing in our build logs, it looks like maybe `hash` is supposed to be a `struct`, not a class. I wonder if that would fix the problem?